Allow nvim as an editor

ehowe opened the pull request:
The regex for determining the editor was a little bit too restrictive. I'm curious why this restriction on editors exists, though. It seems like any valid text editor should work here. I commented this block out and was able to use most other editors I tried. If you are open to it, I think I can make this much more robust.

ehowe committed:
allow nvim as an editor

ehowe committed:
cleanup regex even more

dsa0x commented:
Thank you @ehowe . Could you also edit the returned error?. 

Actually I think I restricted this because I found an editor that it doesn't work with. But I agree we can remove the restrictions and put some caveats about incompatible editors

ehowe committed:
remove editor restriction and enhance gui editor capability

ehowe committed:
update documentation

ehowe commented:
I finally made it back to this. As we discussed, I removed the editor restriction. I also enhanced the GUI editor capability. Finally, I updated the Readme to reflect the changes that were made.

dsa0x commented:
> I finally made it back to this. As we discussed, I removed the editor restriction. I also enhanced the GUI editor capability. Finally, I updated the Readme to reflect the changes that were made.

Thank you @ehowe
